client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -278,6 +280,33 @@ public RssShuffleManager(SparkConf conf, boolean isDriver) {
     this.reassignedFaultyServers = JavaUtils.newConcurrentMap();
   }
 
+  private static void fetchAndApplyDynamicConf(SparkConf sparkConf) {
+    String clientType = sparkConf.get(RssSparkConfig.RSS_CLIENT_TYPE);
+    String coordinators = sparkConf.get(RssSparkConfig.RSS_COORDINATOR_QUORUM.key());
+    CoordinatorClientFactory coordinatorClientFactory = CoordinatorClientFactory.getInstance();
+    List<CoordinatorClient> coordinatorClients =
+        coordinatorClientFactory.createCoordinatorClient(
+            ClientType.valueOf(clientType), coordinators);
+
+    int timeoutMs =
+        sparkConf.getInt(
+            RssSparkConfig.RSS_ACCESS_TIMEOUT_MS.key(),
+            RssSparkConfig.RSS_ACCESS_TIMEOUT_MS.defaultValue().get());
+    for (CoordinatorClient client : coordinatorClients) {
+      RssFetchClientConfResponse response =
```

Comment:
why not using `shuffleWriteClient.fetchClientConf` ?

Reply:
Using `shuffleWriteClient.fetchClientConf` to fetch the dynamic client conf disallows configuring the creation of `ShuffleWriteClient` through dynamic config. See #1554.

Reply:
Got it. But I want to know why re-creating the write client? Can you share some problems that you encountered?

Reply:
Configuring the block id layout via dynamic conf requires this: https://github.com/apache/incubator-uniffle/pull/1528/files#diff-ea644edb1c0bf0e80f9a960adbc1615c99cb6a3a0d5fe24f788307f1daf22f46R127-R128